manim/mobject/geometry.py

```diff
@@ -1428,143 +1432,313 @@ def __init__(self, start_anchor, start_handle, end_handle, end_anchor, **kwargs)
+
+            # To ensure that we loop through starting with last
+            arcs = [arcs[-1], *arcs[:-1]]
+            for arc1, arc2 in adjacent_pairs(arcs):
+                new_points.extend(arc1.get_points())
+
+                line = Line(arc1.get_end(), arc2.get_start())
+
+                # Make sure anchors are evenly distributed
+                len_ratio = line.get_length() / arc1.get_arc_length()
+
+                line.insert_n_curves(int(arc1.get_num_curves() * len_ratio))
+
+                new_points.extend(line.get_points())
+
+        self.set_points(new_points)
 
-        self.clear_points()
-        # To ensure that we loop through starting with last
-        arcs = [arcs[-1], *arcs[:-1]]
-        for arc1, arc2 in adjacent_pairs(arcs):
-            self.append_points(arc1.points)
-            line = Line(arc1.get_end(), arc2.get_start())
-            # Make sure anchors are evenly distributed
-            len_ratio = line.get_length() / arc1.get_arc_length()
-            line.insert_n_curves(int(arc1.get_num_curves() * len_ratio))
-            self.append_points(line.get_points())
         return self
 
 
-class RegularPolygon(Polygon):
-    """An n-sided regular polygon.
+class Polygon(Polygram):
+    """A shape consisting of one closed loop of vertices.
 
     Parameters
     ----------
-    n : :class:`int`
-        The number of sides of the polygon.
-    start_angle : Optional[:class:`float`]
-        The angle at which the polygon is rotated.
-    kwargs : Any
-        Additional arguments to be passed to :class:`Polygon`
+    vertices
+        The vertices of the :class:`Polygon`.
+    kwargs
+        Forwarded to the parent constructor.
+
+    Examples
+    --------
+
+    .. manim:: PolygonExample
+        :save_last_frame:
+
+        class PolygonExample(Scene):
+            def construct(self):
+                isosceles = Polygon([-5, 1.5, 0], [-2, 1.5, 0], [-3.5, -2, 0])
+                position_list = [
+                    [4, 1, 0],  # middle right
+                    [4, -2.5, 0],  # bottom right
+                    [0, -2.5, 0],  # bottom left
+                    [0, 3, 0],  # top left
+                    [2, 1, 0],  # middle
+                    [4, 3, 0],  # top right
+                ]
+                square_and_triangles = Polygon(*position_list, color=PURPLE_B)
+                self.add(isosceles, square_and_triangles)
+    """
+
+    def __init__(self, *vertices: Sequence[float], **kwargs):
+        super().__init__(vertices, **kwargs)
+
+
+class RegularPolygram(Polygram):
+    """A :class:`Polygram` with regularly spaced vertices.
+
+    Parameters
+    ----------
+    num_vertices
+        The number of vertices.
+    density
+        The density of the :class:`RegularPolygram`.
+
+        Can be thought of as how many vertices to hop
+        to draw a line between them. Every ``density``-th
+        vertex is connected.
+    radius
+        The radius of the vertices.
```

Comment:
```suggestion
        The radius of the the circle the vertices are placed on.
```

Reply:
In my mind, this wording feels clunkier and unnecessary. I don't think it's incorrect to say "radius of the vertices", right?

Reply:
To me this sounds like the vertices themselves were circles with their own radius... But I might be nitpicking.